### PR TITLE
Get studies with notifications enabled

### PR DESCRIPTION
**Pull request by anzioka**
PR: filtered search on studies entity set to get studies that have notifications enabled

### CODE REVIEW

**Comment by anzioka** [comment]
Build fails because getQuestionnaireChronicle is not yet in develop

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 1]
```diff
@@ -760,40 +765,60 @@ public boolean isKnownParticipant( UUID studyId, String participantId ) {
...
-                                studyIds.contains( UUID
-                                        .fromString( neighbor.getAssociationDetails().get( OL_ID_FQN ).iterator().next()
-                                                .toString() ) ) )
-                        .count() != 0 )
-                .map( Map.Entry::getKey )
-                .collect( Collectors.toSet() );
+        /*
+         * studies with notifications enabled have 'ol.id' property in the corresponding
+         * associationDetails set to the value of the studyId
+         * For each study, there is can only be at most 1 notification -> partof -> study association,
```
Comment (jokedurnez) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:798]: Isn't this gonna change very soon when there will be a notification for the questionnaire?

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:798]: um, good point. I actually haven't thought about how to do notifications for the questionnaire. Maybe we should have a `notification -> partof -> questionnaire` association?  I'd like to sync about that @jokedurnez

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 2]
```diff
@@ -760,40 +765,60 @@ public boolean isKnownParticipant( UUID studyId, String participantId ) {
...
-                .map( study -> UUID.fromString( study.get( STRING_ID_FQN ).iterator().next().toString() ) )
-                .collect( Collectors.toSet() );
+        //mapping from studyEKID -> studyId
+        Map<UUID, UUID> studyEntityKeyIdMap = new HashMap<>();
+
+        studies.forEach( study -> {
+            UUID studyId = UUID.fromString( study.get( STRING_ID_FQN ).iterator().next().toString() );
+            UUID entityKeyId = UUID.fromString( study.get( ID_FQN ).iterator().next().toString() );
+            studyEntityKeyIdMap.put( entityKeyId, studyId );
+        } );
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:782]: You're already looping over the studies above on line 770. You can merge / simplify this logic:

```
Map<UUID, UUID> idMap = studies
    .stream()
    .collect( Collectors.toMap(
        study -> UUID.fromString( study.get( ID_FQN ).iterator().next().toString() ),
        study -> UUID.fromString( study.get( STRING_ID_FQN ).iterator().next().toString() )
    ) );

Set<UUID> studyEntityKeyIds = idMap.keySet();
```

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:782]: Awesome.

**Pull request merged**